crates/sui-indexer/src/models/addresses.rs

```diff
@@ -1,25 +1,14 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::models::transactions::Transaction;
-use crate::schema::addresses;
-
 use diesel::prelude::*;
 
+use crate::schema::addresses;
+
 #[derive(Queryable, Insertable, Debug)]
 #[diesel(table_name = addresses, primary_key(account_address))]
 pub struct Address {
     pub account_address: String,
     pub first_appearance_tx: String,
     pub first_appearance_time: i64,
 }
-
-impl From<&Transaction> for Address {
```

Comment:
question: so previously we'd derive address from transaction, but now we basically want to lift addresses to be queried directly? Would it still be useful to keep this

Reply:
this PR changed the semantic meaning of an address now, before this PR, it was sender or an txn, but after this PR, it is not sender or recipient account address.

txn has a `recipients` field but it is strings of Owner, which can be addresses owners of other types like
```
pub enum Owner {
    /// Object is exclusively owned by a single address, and is mutable.
    AddressOwner(SuiAddress),
    /// Object is exclusively owned by a single object, and is mutable.
    /// The object ID is converted to SuiAddress as SuiAddress is universal.
    ObjectOwner(SuiAddress),
    /// Object is shared, can be used by any address, and is mutable.
    Shared {
        /// The version at which the object became shared
        initial_shared_version: SequenceNumber,
    },
    /// Object is immutable, and hence ownership doesn't matter.
    Immutable,
}
```